src/codegen/mos6502.rs

```diff
@@ -443,32 +444,42 @@ pub unsafe fn generate_function(name: *const c_char, name_loc: Loc, code_start:
                 store_auto(output, result, asm);
             },
             Op::Asm {args: _} => unreachable!(),
-            // Op::JmpIfNot{addr, arg} => {
-            //     load_arg(arg, op.loc, output, asm);
-
-            //     write_byte(output, CMP_IMM);
-            //     write_byte(output, 0);
-
-            //     // if !=0, skip next check and branch
-            //     write_byte(output, BNE);
-            //     write_byte(output, 7); // skip next 4 instructions
-
-            //     write_byte(output, CPY_IMM);
-            //     write_byte(output, 0);
-
-            //     write_byte(output, BNE);
-            //     write_byte(output, 3);
-
-            //     write_byte(output, JMP_ABS);
-            //     add_reloc(output, RelocationKind::AddressAbs{idx: *op_addresses.items.add(addr)}, asm);
-            // },
-            // Op::Jmp{addr} => {
-            //     write_byte(output, JMP_ABS);
-            //     add_reloc(output, RelocationKind::AddressAbs{idx: *op_addresses.items.add(addr)}, asm);
-            // },
-            Op::Label          {..} => missingf!(op.loc, c!("Label-style IR\n")),
-            Op::JmpLabel       {..} => missingf!(op.loc, c!("Label-style IR\n")),
-            Op::JmpIfNotLabel  {..} => missingf!(op.loc, c!("Label-style IR\n")),
+            Op::Label{label} => {
+                write_byte(output, NOP);
+                // TODO: find a way to avoid using temporary buffer to create a label
+                let label = temp_sprintf(c!(".%s%zu"), name, label);
+                da_append(&mut (*asm).labels, Label {
+                    name: label,
+                    addr: (*output).count as u16,
+                });
+            },
+            Op::JmpLabel{label} => {
+                // TODO: find a way to avoid using temporary buffer to create a label
+                let label = temp_sprintf(c!(".%s%zu"), name, label);
```

Comment:
I don't accept such solution. `struct Label` in mos6502 codegen was clearly designed for function names. I'd suggest looking in the direction of `relocs`.